src/app/moviedb-details/moviedb-details.component.ts

```diff
@@ -1,10 +1,44 @@
 import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { switchMap, Observable, EMPTY, map } from 'rxjs';
+import { MoviedbSeriesDetailsApiService } from './moviedb-series-details-api.service';
+import { Details, ActorCollection, CrewCollection, CreditsResponse, CreditsData } from './moviedb-series-details';
 
 @Component({
   selector: 'app-moviedb-details',
   templateUrl: './moviedb-details.component.html',
-  styleUrls: ['./moviedb-details.component.scss']
+  styleUrls: ['./moviedb-details.component.scss'],
 })
 export class MoviedbDetailsComponent {
+  private data$: Observable<CreditsData> = this.activatedRoute.paramMap.pipe(
+    switchMap((paramMap) => this.getData(paramMap.get('type'), paramMap.get('name'))),
+    map((response) => ({
+      cast: response.cast.map((item) => ({ name: item.name, path: item.profile_path })),
+      crew: response.crew.map((item) => ({ name: item.name })),
+    }))
+  );
 
+  actors$: Observable<ActorCollection[]> = this.data$.pipe(map((data) => data.cast));
+
+  crew$: Observable<CrewCollection[]> = this.data$.pipe(map((data) => data.crew));
+
+  details$: Observable<Details> = this.activatedRoute.paramMap.pipe(
+    switchMap((paramMap) => this.requestDetails(paramMap.get('type'), paramMap.get('name')))
+  );
+
+  constructor(private readonly activatedRoute: ActivatedRoute, private readonly detailsApi: MoviedbSeriesDetailsApiService) {}
+
+  private requestDetails(type: string | null, name: string | null): Observable<Details> {
+    if (!name) {
```

Comment:
It's not safe to define this parameters via url. If I go `http://localhost:4200/blba/bala` now, it will be an empty page with errors in the console.

Let's create to separate routes for series and movies. Then we will guard `:type` param.

We could discuss this moment offline